{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "436c33ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CHROM',\n",
       " 'POS',\n",
       " 'REF',\n",
       " 'ALT',\n",
       " 'AF_ESP',\n",
       " 'AF_EXAC',\n",
       " 'AF_TGP',\n",
       " 'CLNDISDB',\n",
       " 'CLNDISDBINCL',\n",
       " 'CLNDN',\n",
       " 'CLNDNINCL',\n",
       " 'CLNHGVS',\n",
       " 'CLNSIGINCL',\n",
       " 'CLNVC',\n",
       " 'CLNVI',\n",
       " 'MC',\n",
       " 'ORIGIN',\n",
       " 'SSR',\n",
       " 'CLASS',\n",
       " 'Allele',\n",
       " 'Consequence',\n",
       " 'IMPACT',\n",
       " 'SYMBOL',\n",
       " 'Feature_type',\n",
       " 'Feature',\n",
       " 'BIOTYPE',\n",
       " 'EXON',\n",
       " 'INTRON',\n",
       " 'cDNA_position',\n",
       " 'CDS_position',\n",
       " 'Protein_position',\n",
       " 'Amino_acids',\n",
       " 'Codons',\n",
       " 'DISTANCE',\n",
       " 'STRAND',\n",
       " 'BAM_EDIT',\n",
       " 'SIFT',\n",
       " 'PolyPhen',\n",
       " 'MOTIF_NAME',\n",
       " 'MOTIF_POS',\n",
       " 'HIGH_INF_POS',\n",
       " 'MOTIF_SCORE_CHANGE',\n",
       " 'LoFtool',\n",
       " 'CADD_PHRED',\n",
       " 'CADD_RAW',\n",
       " 'BLOSUM62']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('clinvar_conflicting.csv', low_memory=False)\n",
    "df.columns.tolist()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "480b7068",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CLNSIGINCL\n",
       "424754:Likely_pathogenic        2\n",
       "30118:risk_factor               2\n",
       "60702:Pathogenic                2\n",
       "236068:Pathogenic               2\n",
       "7901:Pathogenic                 2\n",
       "                               ..\n",
       "424713:Pathogenic               1\n",
       "3620:Benign                     1\n",
       "424813:Pathogenic               1\n",
       "10382:other                     1\n",
       "10361:Pathogenic|10382:other    1\n",
       "Name: count, Length: 137, dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['CLNSIGINCL'].value_counts()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c62c85cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def simplify_clnsig(x):\n",
    "    x = str(x)\n",
    "    parts = x.split(':')\n",
    "    label = parts[-1]  \n",
    "    label = label.split('|')[0]  \n",
    "    label = label.strip().lower() \n",
    "\n",
    "    if label in ['benign']:\n",
    "        return 0\n",
    "    elif label in ['pathogenic', 'likely_pathogenic']:\n",
    "        return 1\n",
    "    else:\n",
    "        return -1 \n",
    "\n",
    "\n",
    "df['label'] = df['CLNSIGINCL'].apply(simplify_clnsig)\n",
    "\n",
    "\n",
    "df_small = df[df['label'] != -1]\n",
    "\n",
    "\n",
    "df_small[['REF','ALT','label']].head()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
